scripts/sof-testbench-build-profile.sh

```diff
@@ -9,18 +9,32 @@ usage() {
     echo "  -h shows this text"
     echo "  -p <platform> sets platform for scripts/rebuild-testbench.sh"
     echo
+    echo "This script must be run from the sof firmware toplevel or workspace directory"
 }
 
 MODULES_S32="asrc dcblock drc drc_multiband eqfir eqiir gain src tdfb"
 MODULES_S24="aria"
 
-if [ -z "${SOF_WORKSPACE}" ]; then
-    echo "Error: environment variable SOF_WORKSPACE need to be set to top level sof directory"
+# This script can be executed from the sof firmware toplevel or workspace directory
+# First check for the worksapce environment variable
+if [ -z "$SOF_WORKSPACE" ]; then
+    # Environment variable is empty or unset so use default
+    BASE_DIR="$HOME/work/sof"
+else
+    # Environment variable exists and has a value
+    BASE_DIR="$SOF_WORKSPACE"
+fi
+cd "$BASE_DIR"
```

Comment:
cannot we use something like
```
SCRIPTS_DIR=$(dirname "${BASH_SOURCE[0]}")
SOF_DIR=$SCRIPTS_DIR/../
```
like other SOF scripts do to avoid guessing?

Reply:
So we have a bit of complexity today as CI has some hard coded rules (with config outside this repo). Long term trying to move to an environment variable where the scripts could be executed in any directory under ```SOF_WORKSPACE```.
I do like this idea, will make the change.